lib/rule-tester/rule-tester.js

```diff
@@ -443,20 +450,34 @@ class RuleTester {
             ].concat(scenarioErrors).join("\n"));
         }
 
+        /**
+         * get linter from linterMap according to cwd
+         * @param {string} [cwd] cwd
+         * @returns {Linter} Linter instance
+         * @private
+         */
+        function getLinter(cwd) {
+            if (!linterMap.has(cwd)) {
+                const linter = new Linter({ cwd });
 
-        linter.defineRule(ruleName, Object.assign({}, rule, {
+                linterMap.set(cwd, linter);
+            }
+            const linter = linterMap.get(cwd);
 
-            // Create a wrapper rule that freezes the `context` properties.
-            create(context) {
-                freezeDeeply(context.options);
-                freezeDeeply(context.settings);
-                freezeDeeply(context.parserOptions);
+            linter.defineRule(ruleName, Object.assign({}, rule, {
 
-                return (typeof rule === "function" ? rule : rule.create)(context);
-            }
-        }));
+                // Create a wrapper rule that freezes the `context` properties.
+                create(context) {
+                    freezeDeeply(context.options);
+                    freezeDeeply(context.settings);
+                    freezeDeeply(context.parserOptions);
 
-        linter.defineRules(this.rules);
+                    return (typeof rule === "function" ? rule : rule.create)(context);
+                }
+            }));
+            linter.defineRules(rules);
```

Comment:
Thank you for the update!

Originally, the tester runs this process one time. Therefore, I think we can move these lines into `if (!linterMap.has(cwd)) { ... }` block.

Reply:
I think the rules won't be set if the tester is ran multiple times.
Say
```
ruleTester.run('rule-a', ...);
ruleTester.run('rule-b', ...);
```
This is because `linterMap.has(cwd)` gives `true` at the second time.

Reply:
Oh, I got it. Thank you for the explanation.

How about clearing `linterMap` at the top of the `run()` method?

My concern is low performance because `getLinter()` will be call on every test pattern. I know several rules that have +10k test patterns.

Reply:
Will do